### ISSUE TITLE
Bar chart x-axis positions can cause different spacing depending on the dates returned

### ISSUE THREAD

**Issue by dah33** [metabase/metabase#14605]
This query:

```
SELECT date_add('days',0,'2021-02-08') as created_at, 2.0 as score
UNION ALL
SELECT date_add('days',0,'2021-02-15'), 3.0
UNION ALL
SELECT date_add('days',0,'2021-02-22'), 1.0
```

Produces fat bars for each week as expected, as each date is 7 days apart:

![image](https://user-images.githubusercontent.com/17231347/106472485-9f94c380-649a-11eb-854a-ff4392c3e65e.png)

But changing the `0` offset for `-8` or lower:

```
SELECT date_add('days',-8,'2021-02-08') as created_at, 2.0 as score
UNION ALL
SELECT date_add('days',-8,'2021-02-15'), 3.0
UNION ALL
SELECT date_add('days',-8,'2021-02-22'), 1.0
```

Makes skinny bars:

![image](https://user-images.githubusercontent.com/17231347/106472657-ce129e80-649a-11eb-80c6-5138fd72dd0f.png)

The same thing happens if you remove the `date_add`, which I was using for convenience of testing.

```
SELECT '2021-01-31' as created_at, 2.0 as score
UNION ALL
SELECT '2021-02-07', 3.0
UNION ALL
SELECT '2021-02-14', 1.0
```

Note, my week start day is Monday in localisation. 

I thought it might be an issue with month boundaries, but adding a positive offset did not seem to produce an issue in my testing.

```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.14.198-152.320.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "redshift"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.3"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-12-03",
      "tag": "v0.37.3",
      "branch": "release-x.37.x",
      "hash": "2f1e783"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Comment by dah33** [comment]
Probably related to #3842?

**Comment by flamber** [comment]
Hi @dah33 
Very related to #3842 - see the [comment](https://github.com/metabase/metabase/issues/3842#issuecomment-279057217) about the calculation. Workaround is the use X-axis scale = Ordinal.
Also, start of week does not apply to SQL:
> First day of the week: This will affect things like grouping by week or filtering in GUI queries. It won't affect SQL queries.

**Comment by paulrosenzweig** [comment]
Fixed by #14706
